### DISCUSSION TITLE
How can I use Flecs query to support the iteration for std algorithms?

### DISCUSSION THREAD

**Discussion by undersilence** [SanderMertens/flecs#900] [Q&A] [answered]
Currently I am working on writing a physics simulator using `Flecs` as internal data container. The whole program can regard as that we have lots of particle entities.
Now, I want my program to support multi-threading (try to use tbb). To brevify, I want to implement the following operations.
```c++
auto Q = world.query<Position, Velocity, Force>();
for(auto it = Q.begin(); it != Q.end(); ++it) {
  auto& [p, v, f] = *it; // Position& p, Velocity& v, Force& f 
  ... 
}
```
So that I can access the result of the query as it was a container. I notice that `query.worker(...)` can divide the results into several parts and iterate them with `.Each(Func&&)` separatly, but I hope there was a more general iteration support.

**Comment by SanderMertens** [comment, marked as answer]
Yeah an interface like that would be useful. Not sure when I'll get to it, but I'll look into it!
